content/storage/docs/spark/instance-store.md

```diff
@@ -0,0 +1,146 @@
+
+**Example**
+
+```
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+  name: YOUR_CLUSTER_NAME
+  region: YOUR_REGION
+
+managedNodeGroups:
+  - name: ng-c5d-9xlarge
+    instanceType: c5d.9xlarge
+    desiredCapacity: 1
+    privateNetworking: true
+    subnets:
+      - YOUR_NG_SUBNET
+    preBootstrapCommands: # commands executed as root
+      - yum install -y mdadm nvme-cli
+      - nvme_disks=($(nvme list | grep "Amazon EC2 NVMe Instance Storage" | awk -F'[[:space:]][[:space:]]+' '{print $1}')) && [[ ${#nvme_disks[@]} -eq 1 ]] && mkfs.ext4 -F ${nvme_disks[*]} && systemctl stop docker && mkdir -p /var/lib/kubelet/pods && mount ${nvme_disks[*]} /var/lib/kubelet/pods && chmod 750 /var/lib/docker && systemctl start docker
+      - nvme_disks=($(nvme list | grep "Amazon EC2 NVMe Instance Storage" | awk -F'[[:space:]][[:space:]]+' '{print $1}')) && [[ ${#nvme_disks[@]} -ge 2 ]] && mdadm --create --verbose /dev/md0 --level=0 --raid-devices=${#nvme_disks[@]} ${nvme_disks[*]} && mkfs.ext4 -F /dev/md0 && systemctl stop docker && mkdir -p /var/lib/kubelet/pods && mount /dev/md0 /var/lib/kubelet/pods && chmod 750 /var/lib/docker && systemctl start docker
+```
+
+
+**Benefits**
+
+* No need to mount the disk using Spark configurations or pod templates
+* Data generated by the application, will immediately be deleted at the pod termination. Data will be also purged in case of pod failures.
+* One time configuration for the node group
+
+**Cons**
+
+* Contention of disk resources if multiple jobs are allocated on the same node, as it’s not possible to properly re-partition resources between jobs
+
+
+
+## **Mount NVMe disks as data volumes**
+
+In this section, we’re going to explicitly mount each instance store volume on the host node, as mount path in the Spark pods (executors and driver).
+
+As in the previous example, this script will automatically format the instance store volumes and create an **xfs** partition. The disks are then mounted in local folders called **/spark_data_IDX** where IDX is an integer that corresponds to the disk mounted.
+
+**Example**
+
+```
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+  name: YOUR_CLUSTER_NAME
+  region: YOUR_REGION
+
+managedNodeGroups:
+  - name: ng-m5d-4xlarge
+    instanceType: m5d.4xlarge
+    desiredCapacity: 1
+    privateNetworking: true
+    subnets:
+      - YOUR_NG_SUBNET
+    preBootstrapCommands: # commands executed as root
+      - "IDX=1;for DEV in /dev/nvme[1-9]n1;do mkfs.xfs ${DEV}; mkdir -p /spark_data_${IDX}; echo ${DEV} /spark_data_${IDX} xfs defaults,noatime 1 2 >> /etc/fstab; IDX=$((${IDX} + 1)); done"
+      - "mount -a"
+      - "chown 999:1000 /spark_data_*"
+```
+
+
+In order to successfully use these volumes with Spark, it’s required to specify additional configurations to use these volumes during shuffles operations. To use a volume as local storage, the volume’s name should starts with `spark-local-dir-`.
```

Comment:
In order to successfully use ephemeral volumes within Spark, you need to specify additional configurations. In addition to spark configuration, the mounted volume name should start with `spark-local-dir-`.